src/firefox/preferences.js

```diff
@@ -117,3 +119,38 @@ export function getPrefs(
     ...appPrefs,
   };
 }
+
+export function coerceCLICustomPreference(
+  cliPrefs: string | Array<string>
+): FirefoxPreferences {
+  let customPrefs = {};
+
+  if (!Array.isArray(cliPrefs)) {
+    cliPrefs = [cliPrefs];
+  }
+
+  for (let pref of cliPrefs) {
+    let [key, value] = pref.split('=');
+
+    if (/[^\w{@}.-]/.test(key)) {
+      throw new UsageError(`Invalid custom preference name: ${key}`);
+    }
+
+    if (value === `${parseInt(value)}`) {
+      value = parseInt(value, 10);
+    } else if (value === 'true' || value === 'false') {
+      value = (value === 'true');
+    }
+
+    let defaultProps = ['devtools.debugger.remote-enabled',
```

Comment:
This can be defined using `const`, I would also rename it to make it immediately clear its role (e.g. something around "non overridable pref names" or "blacklisted custom pref names") and move out of this function, in the first part of this module (as an exported module const, which would also be helpful for the related tests).

Also, this array should be formatted like in:

```
const varName = [
  ..., // max 80 cols
  ..., // go to the new line if the line above is already too long 
  ..., // and add a "dangle comma" at the end (which prevents syntax errors if/when 
  ..., // a new element is appended).  
];
```